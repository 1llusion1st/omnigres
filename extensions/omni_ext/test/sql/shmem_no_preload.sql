-- Since it's not preloaded, this test should not pass
SELECT omni_ext_test_no_preload.alloc_shmem_global();
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();

-- Load
SELECT omni_ext.load('omni_ext_test_no_preload');

-- Should work now
SELECT omni_ext_test_no_preload.alloc_shmem_global();
-- Should work now, too
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();
CREATE DATABASE another_db;
\connect another_db
CREATE EXTENSION omni_ext_test_no_preload CASCADE;
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();