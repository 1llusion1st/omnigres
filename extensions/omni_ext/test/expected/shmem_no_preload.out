-- Since it's not preloaded, this test should not pass
SELECT omni_ext_test_no_preload.alloc_shmem_global();
ERROR:  no allocation found
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();
ERROR:  no allocation found
-- Load
SELECT omni_ext.load('omni_ext_test_no_preload');
NOTICE:  _Dynpgext_init
 load 
------
 0.1
(1 row)

-- Should work now
SELECT omni_ext_test_no_preload.alloc_shmem_global();
 alloc_shmem_global 
--------------------
 test
(1 row)

-- Should work now, too
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();
 alloc_shmem_database_local 
----------------------------
 testdb 0
(1 row)

CREATE DATABASE another_db;
\connect another_db
CREATE EXTENSION omni_ext_test_no_preload CASCADE;
NOTICE:  installing required extension "omni_ext"
SELECT omni_ext_test_no_preload.alloc_shmem_database_local();
 alloc_shmem_database_local 
----------------------------
 testdb 1
(1 row)

